src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverter.cs

```diff
@@ -25,6 +25,14 @@ public abstract partial class JsonConverter
 
         internal abstract ConverterStrategy ConverterStrategy { get; }
 
+        /// <summary>
+        /// Indicates that the converter can consume the <see cref="JsonTypeInfo.CreateObject"/> delegate.
+        /// Needed because certain collection converters cannot support arbitrary delegates.
+        /// TODO remove once https://github.com/dotnet/runtime/pull/73395/ and
+        /// https://github.com/dotnet/runtime/issues/71944 have been addressed.
+        /// </summary>
+        internal virtual bool SupportsCreateObjectDelegate => false;
```

Comment:
A bunch of the overrides seem to explicitly return false.  Why are those needed if this is virtual and the base returns false?  Shouldn't we only override if we need to return something else?  Or is it the case for all of those that their base type is overriding this to return something other than false, and so they need to override to go back to this default?

Reply:
There are a couple of intermediate classes in the hierarchy, namely `JsonCollectionConverter` and `JsonDictionaryConverter` whose implementation of the `CreateCollection` methods make use of the `CreateObject` delegate, however there's a number of derived types that override this method and ignore the delegate completely. It felt reasonable to me that the `SupportsCreateObject` overrides should reflect whatever the current `CreateCollection` override is doing.